pkg/diagnostics/grpc_tracing.go

```diff
@@ -47,17 +47,11 @@ func GRPCTraceUnaryServerInterceptor(appID string, spec config.TracingSpec) grpc
 
 		ctx, span = trace.StartSpanWithRemoteParent(ctx, spanName, sc, sampler, spanKind)
 
-		var prefixedMetadata map[string]string
-		if span.SpanContext().TraceOptions.IsSampled() {
-			// users can add dapr- prefix if they want to see the header values in span attributes.
-			prefixedMetadata = userDefinedMetadata(ctx)
-		}
-
 		resp, err := handler(ctx, req)
 
 		if span.SpanContext().TraceOptions.IsSampled() {
 			// Populates dapr- prefixed header first
-			AddAttributesToSpan(span, prefixedMetadata)
+			AddAttributesToSpan(span, userDefinedMetadata(ctx))
```

Comment:
please revert this code. what happens if ctx has been changed after line 56?

Reply:
ctx is being passed as value right, not reference so we won't be able to capture the changes unlike the http side

Reply:
I was wrong actually. It's indeed holding the reference. Reverted this change.